src/ol/renderer/Map.js

```diff
@@ -250,15 +251,14 @@ class MapRenderer extends Disposable {
   }
 
   /**
+   * @param {Array<string>} layerKeys Layer keys to remove.
    * @param {import("../PluggableMap.js").default} map Map.
    * @param {import("../PluggableMap.js").FrameState} frameState Frame state.
    * @private
    */
-  removeUnusedLayerRenderers_(map, frameState) {
-    for (const layerKey in this.layerRenderers_) {
-      if (!frameState || !(layerKey in frameState.layerStates)) {
-        this.removeLayerRendererByKey_(layerKey).dispose();
-      }
+  removeUnusedLayerRenderers_(layerKeys, map, frameState) {
+    for (const layerKey in layerKeys) {
```

Comment:
Looks like `layerKey` is going to be the array index here (instead of the layer key).